src/CpmsClient/Service/ApiService.php

```diff
@@ -7,90 +8,91 @@
-    const SCOPE_CHEQUE       = 'CHEQUE';
-    const SCOPE_REFUND       = 'REFUND';
-    const SCOPE_QUERY_TXN    = 'QUERY_TXN';
-    const SCOPE_STORED_CARD  = 'STORED_CARD';
-    const SCOPE_CHARGE_BACK  = 'CHARGE_BACK';
-    const SCOPE_CASH         = 'CASH';
-    const SCOPE_POSTAL_ORDER = 'POSTAL_ORDER';
-    const SCOPE_CHIP_PIN     = 'CHIP_PIN';
-    const SCOPE_ADJUSTMENT   = 'ADJUSTMENT';
-    const SCOPE_REPORT       = 'REPORT';
-    const CHEQUE_RD          = 'CHEQUE_RD'; // refer to drawer
-    const DIRECT_DEBIT_IC    = 'DIRECT_DEBIT_IC'; // indemnity claim
-    const REALLOCATE_PAYMENT = 'REALLOCATE'; // Reallocate payments by switch customer reference
-    const MAX_RETIRES        = 3;
-    /**
-     * @var LoggerInterface
-     */
-    protected $logger = null;
+    public const SCOPE_CARD         = 'CARD';
+    public const SCOPE_CNP          = 'CNP';
+    public const SCOPE_DIRECT_DEBIT = 'DIRECT_DEBIT';
+    public const SCOPE_CHEQUE       = 'CHEQUE';
+    public const SCOPE_REFUND       = 'REFUND';
+    public const SCOPE_QUERY_TXN    = 'QUERY_TXN';
+    public const SCOPE_STORED_CARD  = 'STORED_CARD';
+    public const SCOPE_CHARGE_BACK  = 'CHARGE_BACK';
+    public const SCOPE_CASH         = 'CASH';
+    public const SCOPE_POSTAL_ORDER = 'POSTAL_ORDER';
+    public const SCOPE_CHIP_PIN     = 'CHIP_PIN';
+    public const SCOPE_ADJUSTMENT   = 'ADJUSTMENT';
+    public const SCOPE_REPORT       = 'REPORT';
+    public const CHEQUE_RD          = 'CHEQUE_RD'; // refer to drawer
+    public const DIRECT_DEBIT_IC    = 'DIRECT_DEBIT_IC'; // indemnity claim
+    public const REALLOCATE_PAYMENT = 'REALLOCATE'; // Reallocate payments by switch customer reference
+    public const MAX_RETIRES        = 3;
 
-    /** @var  \Laminas\Cache\Storage\StorageInterface */
-    protected $cacheStorage;
-    /**
-     * @var \CpmsClient\Client\HttpRestJsonClient
-     */
-    protected $client;
+    protected LoggerInterface $logger;
+
+    protected StorageInterface $cacheStorage;
 
-    /** @var  \Laminas\ServiceManager\ServiceManager */
-    protected $serviceManager;
+    protected HttpRestJsonClient $client;
 
-    /** @var array */
-    protected $tokens = array();
+    protected array $tokens = [];
 
-    /** @var ClientOptions */
-    protected $options;
+    protected ClientOptions $options;
 
-    /** @var bool */
-    protected $enableCache = true;
+    protected bool $enableCache = true;
 
-    /** @var \DVSA\CPMS\Queues\QueueAdapters\Queues */
-    protected $queuesClient;
+    protected NotificationsClient $queuesClient;
 
     // we need to refactor the code to put these in a common package
     // that can be shared by both the client and the server :(
-    const CPMS_CODE_SUCCESS = '000';
+    public const CPMS_CODE_SUCCESS = '000';
 
     /**
      * Number of retries to get a valid token
-     *
-     * @var int
      */
-    private static $retries = 0;
+    private static int $retries = 0;
+
+    public function __construct(
```

Comment:
Will this construct change the behaviour? if not can we use here "Constructor property promotion"?

Reply:
Have changed to constructor property promotion